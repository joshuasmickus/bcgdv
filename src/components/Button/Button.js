import React from 'react';
import { func, node, oneOfType, string } from 'prop-types';


export const Button = ({ className, onClick, children }) => (
  <button className={`button ${className}`} onClick={onClick}>{children}</button>
);


Button.propTypes = {
  className: string,
  onClick: func.isRequired,
  children: oneOfType([
    string,
    node
  ]).isRequired
};

Button.defaultProps = {
  className: ''
}
