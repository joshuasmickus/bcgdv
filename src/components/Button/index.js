import './Button.css';

export { Button as default } from './Button';
