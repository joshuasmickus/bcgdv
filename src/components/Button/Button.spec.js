import React from 'react';
import { shallow } from 'enzyme';

import { Button } from './Button';

const onClickMock = jest.fn();
const requiredProps = {
  onClick: onClickMock,
  children: 'here'
};

describe('Given a Button component', () => {
  describe('when it is rendered', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<Button {...requiredProps} />);

      expect(wrapper.getElement()).toMatchSnapshot();
    });

    describe('and the button is clicked', () => {
      it('should trigger the onClick prop', () => {
        const wrapper = shallow(<Button {...requiredProps} />);

        wrapper.find('button').simulate('click');

        expect(onClickMock).toHaveBeenCalledWith();
      });
    });
  });
});
