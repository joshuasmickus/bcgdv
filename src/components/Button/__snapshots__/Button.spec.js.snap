// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Button component when it is rendered should match the snapshot 1`] = `
<button
  className="button "
  onClick={[MockFunction]}
>
  here
</button>
`;
